src/wrappers.jl

```diff
@@ -145,18 +145,18 @@ function SciMLBase.solve(cache::LinearCache, alg::KrylovJL; kwargs...)
     kwargs = (atol=abstol, rtol=reltol, itmax=maxiter, verbose=verbose,
               alg.kwargs...)
 
-    if cache.cacheval isa Krylov.CgSolver
+    if alg.KrylovAlg === Krylov.cg!
```

Comment:
isn't this computed at compile time already?

Reply:
cache.cacheval is set at runtime in init_cacheval().

since KrylovAlg is known before and is used to set KrylovSolver, branching on KrylovAlg would be faster. what do you think?

Reply:
what does inference say? I assume this is fully inferred at compile time.